tests/foreman/sys/test_hooks.py

```diff
@@ -25,137 +25,130 @@
-        self.assertEqual(result.return_code, 0)
-        self.assertIn(expected_msg.format(destroy_event, host_name), result.stdout[0])
-
-    @pytest.mark.destructive
-    def test_positive_hostgroup_hooks(self):
-        """Create hooks to be executed on hostgroup create, udpdate and destroy
-
-        :id: 7e935dec-e4fe-47d8-be02-8c687a99ae7a
-
-        :steps:
-            1. Create hook directories with symlinks to logger script
-            for subset of supported events: before_create, before_update,
-            before_destroy
-            2. Perform trigger actions (create, update and delete host group)
-            3. Observe custom logs created by hook script
-
-        :expectedresults: hook scripts are executed at proper time
-        """
-        hg_name = valid_hostgroups_list()[0]
-        expected_msg = "Executed {0} hook on object {1}"
-
-        # create hook destination directories with logger script
-        create_event = "before_create"
-        update_event = "before_update"
-        destroy_event = "before_destroy"
-        for event in [create_event, update_event, destroy_event]:
-            hook_dir = f"{HOOKS_DIR}/hostgroup/{event}"
-            ssh.command(f'''mkdir -p {hook_dir}''')
-            ssh.command(f'''ln -sf {self.script_path} {hook_dir}/''')
-        result = ssh.command("systemctl restart httpd")
-        self.assertEqual(result.return_code, 0)
-
-        # create hg, check logs for hook activity
-        hg = entities.HostGroup(name=hg_name, organization=[self.org]).create()
-        self.assertEqual(hg.name, hg_name)
-        result = ssh.command(f'cat {LOGS_DIR}')
-        self.assertEqual(result.return_code, 0)
-        self.assertIn(expected_msg.format(create_event, hg_name), result.stdout[0])
-
-        # update hg, check logs for hook activity
-        new_arch = entities.Architecture().create()
-        hg.architecture = new_arch
-        hg = hg.update(['architecture'])
-        self.assertEqual(hg.architecture.read().name, new_arch.name)
-        result = ssh.command(f'cat {LOGS_DIR}')
-        self.assertEqual(result.return_code, 0)
-        self.assertIn(expected_msg.format(update_event, hg_name), result.stdout[0])
-
-        # delete hg, check logs for hook activity
-        hg.delete()
-        with self.assertRaises(HTTPError):
-            hg.read()
-        result = ssh.command(f'cat {LOGS_DIR}')
-        self.assertEqual(result.return_code, 0)
-        self.assertIn(expected_msg.format(destroy_event, hg_name), result.stdout[0])
+SCRIPT_PATH = f"{HOOKS_DIR}/logger.sh"
+
+
+pytestmark = [pytest.mark.run_in_one_thread, pytest.mark.destructive]
+
+
+@pytest.fixture(scope='function')
+def logger_hook(default_org):
+    """Create logger script to be executed via hooks"""
+    ssh.command(
+        '''printf '#!/bin/sh\necho "$(date): Executed $1 hook'''
+        + f''' on object $2" > {LOGS_DIR}' > {SCRIPT_PATH}'''
+    )
+    ssh.command(f"chmod 774 {SCRIPT_PATH}")
```

Comment:
We've still got to prefer single over double quotes

Reply:
ack, this is a good time to update them. I changed them where I was modifying lines, will get the rest in this file in this branch.